aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -143,6 +193,161 @@ const char* _cublasGetErrorEnum(cublasStatus_t error) {
     CUDABLAS_POSINT_CHECK(gemm<Dtype>, ldc);  \
   } while (0)
 
+#define BGEMM_CHECK_ARGVALUES(Dtype)           \
+  do {                                        \
+    CUDABLAS_NONNEGINT_CHECK(bgemm<Dtype>, m); \
+    CUDABLAS_NONNEGINT_CHECK(bgemm<Dtype>, n); \
+    CUDABLAS_NONNEGINT_CHECK(bgemm<Dtype>, k); \
+    CUDABLAS_POSINT_CHECK(bgemm<Dtype>, lda);  \
+    CUDABLAS_POSINT_CHECK(bgemm<Dtype>, ldb);  \
+    CUDABLAS_POSINT_CHECK(bgemm<Dtype>, ldc);  \
+    CUDABLAS_POSINT_CHECK(bgemm<Dtype>, num_batches);  \
```

Comment:
This is now failing:
```python
import torch

a = torch.randn((0, 5, 5), device='cuda')
torch.bmm(a, a)
```
with
```
Traceback (most recent call last):
  File "bmm.py", line 4, in <module>
    torch.bmm(a, a)
RuntimeError: at::cuda::blas::bgemm<float> argument num_batches must be positive and less than 2147483647 but got 0
Exception raised from bgemm<float> at ../aten/src/ATen/cuda/CUDABlas.cpp:228 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x5c (0x7f932eef471c in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libc10.so)
frame #1: <unknown function> + 0xb8e17e (0x7f932faaa17e in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #2: at::native::(anonymous namespace)::baddbmm_out_cuda_impl(at::Tensor&, at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::Scalar, c10::Scalar) + 0x1227 (0x7f93309cfbf7 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #3: at::native::bmm_out_cuda(at::Tensor&, at::Tensor const&, at::Tensor const&) + 0xc2 (0x7f93309d0472 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #4: at::native::bmm_cuda(at::Tensor const&, at::Tensor const&) + 0x6c (0x7f93309d057c in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #5: <unknown function> + 0x28c8919 (0x7f93317e4919 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #6: <unknown function> + 0x2924c34 (0x7f9331840c34 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #7: <unknown function> + 0x17ee197 (0x7f935f18e197 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #8: at::bmm(at::Tensor const&, at::Tensor const&) + 0xb1 (0x7f935f192b11 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #9: <unknown function> + 0x2bc1404 (0x7f9360561404 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #10: <unknown function> + 0x2bc1774 (0x7f9360561774 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #11: <unknown function> + 0x19d6837 (0x7f935f376837 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #12: at::Tensor::bmm(at::Tensor const&) const + 0xb1 (0x7f935f379b21 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #13: <unknown function> + 0x609548 (0x7f9364f91548 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_python.so)
<omitting python frames>
frame #25: __libc_start_main + 0xf2 (0x7f9396b73152 in /usr/lib/libc.so.6)
```

Reply:
oh I didn't know we support 0 batch size. changed it to `CUDABLAS_NONNEGINT_CHECK`

Reply:
I will write a separate PR to add a test for this case

Reply:
The PR for testing this case is at https://github.com/pytorch/pytorch/pull/47700, I actually found some bugs, and fixed them.

Reply:
Let's land https://github.com/pytorch/pytorch/pull/47700 first, and rebase and test this PR with it.